src/BuildHistory.test.tsx

```diff
@@ -0,0 +1,156 @@
+import "./__mocks__/matchMedia.mock.js";
+
+import React from "react";
+
+import {fireEvent, getByText, getByTitle, prettyDOM, render, screen} from "@testing-library/react";
+import BuildContext, {Build, Metadata} from "./BuildContext";
+import BuildHistory from "./BuildHistory";
+import {v4 as uuid_v4} from "uuid";
+
+describe("test BuildHistory", () => {
+    const construct = () => {
+        const data = {
+            id: uuid_v4(),
+            name: "A Build Worth Naming",
+            createdAt: new Date(),
+            updatedAt: new Date()
+        }
+        const history = [data] as Metadata[]
+
+        const remove = jest.fn() as (id: string) => void
+        const removeAll = jest.fn() as () => void
+
+        const wrapper = render(
+            <BuildContext.Provider value={{history, remove, removeAll} as Build}>
+                <BuildHistory/>
+            </BuildContext.Provider>
+        )
+
+        return {
+            wrapper,
+            data,
+            build: {history, remove, removeAll}
+        }
+    }
+
+    const parentOf = (element: HTMLElement): HTMLElement => {
+        const {parentElement} = element
+        expect(parentElement).toBeInTheDocument()
+        if (parentElement === null) fail()
+        return parentElement
+    }
+
+    it("should render with history present", async () => {
+        construct()
+
+        const h3 = screen.getByText(/^History$/)
+        expect(h3).toBeInTheDocument()
+        expect(h3.tagName).toBe("H3")
+    })
+
+    it("should have rows with history present", async () => {
+        const {build: {history}} = construct()
+
+        const list = screen.getByTitle("BuildHistory-list")
+        expect(list).toBeInTheDocument()
+        expect(list.childElementCount).toEqual(history.length)
+    })
+
+    it("should have a row with expected data", async () => {
+        const {data} = construct()
+
+        const h3 = screen.getByText(/^History$/)
+        expect(h3).toBeInTheDocument()
+        expect(h3.tagName).toBe("H3")
```

Comment:
I don't know if these assertions need to be in every test. Certainly not the `tagName` assertion